src/hooks/useMaciSignup.ts

```diff
@@ -44,12 +52,23 @@ export const useMaciSignup = (): IUseMaciSignUpData => {
       return;
     }
 
-    const { stateIndex: index, hash } = await signup({
-      maciPubKey: data.publicKey,
-      maciAddress: config.maciAddress!,
-      sgDataArg: attestationId,
-      signer: signer as unknown as Signer,
-    });
+    setIsLoading(true);
+
+    let index, hash;
+
+    try {
+      const { stateIndex, hash: _hash } = await signup({
+        maciPubKey: data.publicKey,
+        maciAddress: config.maciAddress!,
+        sgDataArg: attestationId,
+        signer: signer as unknown as Signer,
+      });
+      index = stateIndex;
+      hash = _hash;
+    } catch (e) {
+      options.onError();
+    }
+    setIsLoading(false);
```

Comment:
It should be added to finally block

Reply:
```
try {
  // signup
} catch (e) {
 // handle error
} finally {
 // set loading
}
```

Reply:
ah the problem is if it's onError, then the button remain to be `loading...`, that's why I put the `setIsLoading` outside. I want the user to 
1. click on signup button (button display **signup**) 
2. waiting for transaction (button display **loading**) 
3. if the transaction succeeded, then display **view ballot**
4. if the transaction failed, then toast error, meanwhile the button display **signup** (for user to try again)

Reply:
Finally block will be called anyway if it's error or not.

Reply:
yep I suddenly think of it so I delete my comment :)